sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/KeyEncryptionKeyAsyncClient.java

```diff
@@ -27,18 +28,17 @@
      */
     KeyEncryptionKeyAsyncClient(String keyId, HttpPipeline pipeline, CryptographyServiceVersion version) {
         super(keyId, pipeline, version);
+        this.keyId = keyId;
```

Comment:
Since you are passing this to the superclass constructor, why not have a superclass manage the lifecycle of this variable? `CryptographyAsyncClient` can have a protected method `getKeyId()` to get this key id.

Reply:
This needs to be Public API only in KEK client.
So moved the variable management to super class as package private method.
Kept it public in KEK client.